src/middleware/errorHandler.js

```diff
@@ -151,67 +175,71 @@ const filterSecrets = (error, req, res, next) => {
 		req.body = filter(req.body);
 		error.data = filter(error.data);
 		error.options = filter(error.options);
+		error.params = filter(error.params);
 	}
 	next(error);
 };
 
-const saveResponseFilter = (error) => ({
-	name: error.name,
-	message: error.message instanceof Error && error.message.message ? error.message.message : error.message,
-	code: error.code,
-	traceId: error.traceId,
-});
+const createErrorDetailTO = (code, type, title, message, customFields = {}) => {
+	return { code, type, title, message, ...customFields };
+};
 
-const sendError = (res, error) => {
-	res.status(error.code).json(saveResponseFilter(error));
+const getErrorResponseFromBusinessError = (businessError) => {
+	const customFields = {};
+	let code = 500;
+	const { message: type, title, defaultMessage: message } = businessError;
+	if (businessError instanceof ValidationError || businessError instanceof AssertionError) {
+		code = 400;
+		Object.assign(customFields, { validation_errors: businessError.params });
+	} else if (businessError instanceof DocumentNotFound) {
+		code = 404;
+	}
+	return createErrorDetailTO(code, type, title, message, customFields);
 };
 
-const handleSilentError = (error, req, res, next) => {
-	if (error instanceof SilentError || (error && error.error instanceof SilentError)) {
-		if (Configuration.get('SILENT_ERROR_ENABLED')) {
-			res.append('x-silent-error', true);
+const getErrorResponse = (error, req, res, next) => {
+	let errorDetail;
+	if (isSilentError(error)) {
+		if (Configuration.get('SILENT_ERROR_ENABLED') === true) {
+			res.append('x-silent-error', true); // TODO is removed in production?
 		}
-		res.status(200).json({ success: 'success' });
+		// do not return this as error via REST
+		return res.status(200).json(SilentError.RESPONSE_CONTENT);
+	}
+	if (isApplicationError(error)) {
+		// Application Errors
+		errorDetail = getErrorResponseFromBusinessError(error);
+	} else if (isFeatherError(error)) {
+		// Framework Errors
+		const { code, className: type, name: title, message } = error;
+		errorDetail = createErrorDetailTO(code, type, title, message);
 	} else {
-		next(error);
+		// Unhandled Errors
+		const unhandledError = new InternalServerError(error);
```

Comment:
InternalServerError is an application or an service error?

Reply:
Application Error